internal/extproc/processor.go

```diff
@@ -204,11 +224,28 @@ func (p *Processor) ProcessResponseBody(_ context.Context, body *extprocv3.HttpB
 	p.costs.InputTokens += tokenUsage.InputTokens
 	p.costs.OutputTokens += tokenUsage.OutputTokens
 	p.costs.TotalTokens += tokenUsage.TotalTokens
+
+	// Track token usage metrics
+	if tokenUsage.InputTokens > 0 {
+		metrics.TokensTotal.WithLabelValues(p.modelName, "prompt").Add(float64(tokenUsage.InputTokens))
+	}
+	if tokenUsage.OutputTokens > 0 {
+		metrics.TokensTotal.WithLabelValues(p.modelName, "completion").Add(float64(tokenUsage.OutputTokens))
+	}
+	if tokenUsage.TotalTokens > 0 {
+		metrics.TokensTotal.WithLabelValues(p.modelName, "total").Add(float64(tokenUsage.TotalTokens))
+	}
+
 	if body.EndOfStream && len(p.config.requestCosts) > 0 {
 		resp.DynamicMetadata, err = p.maybeBuildDynamicMetadata()
 		if err != nil {
 			return nil, fmt.Errorf("failed to build dynamic metadata: %w", err)
 		}
+		if p.backendName != "" && p.modelName != "" {
```

Comment:
also here - when exactly does this happen? if it ever happens then we need a code comment here as it's not obvious without the context vs if not let's remove. Either way this is not tested

Reply:
removed